ql/src/java/org/apache/hadoop/hive/ql/ddl/table/drop/DropTableAnalyzer.java

```diff
@@ -46,9 +48,16 @@ public DropTableAnalyzer(QueryState queryState) throws SemanticException {
 
   @Override
   public void analyzeInternal(ASTNode root) throws SemanticException {
-    String tableName = getUnescapedName((ASTNode) root.getChild(0));
+    TableName qualTabName = getQualifiedTableName((ASTNode) root.getChild(0));
+    String tableName = qualTabName.getNotEmptyDbTable();
     boolean ifExists = (root.getFirstChildWithType(HiveParser.TOK_IFEXISTS) != null);
     boolean throwException = !ifExists && !HiveConf.getBoolVar(conf, ConfVars.DROP_IGNORES_NON_EXISTENT);
+    //Authorize database for drop table command
+    // Skip db object if database doesn't exist
+    Database database  = getDatabase(qualTabName.getDb(),false);
```

Comment:
@zabetak  In case of the DROP TABLE IF EXISTS command, this `throwException` refers to throwing exception in case of a non-existent table during the `getTable` parameter. As we want NOOP in case of non-existent DB in DROP TABLE, hardcoded this as  false for `getDatabase` call